src/common/actions/register-name.js

```diff
@@ -198,3 +211,79 @@ export function registerNameClear(id) {
     }
   };
 }
+
+export async function internalNameOwnership(root, discharge, snapName) {
+  if (!snapName) {
+    throw new APICompatibleError({
```

Comment:
Should this be a APICompatibleError given the source of this error is from the application and not the API? A standard `Error` would probably do.